### PR TITLE
ON-44411 # Changed externalId generation to be from API

### PR DESCRIPTION
**Pull request by jdawg093**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Updated
```
Comment (mymattcarroll) [inline CHANGELOG.md:10]:
```suggestion
### Changed
```

We want to match the format described here: https://keepachangelog.com/en

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+
+- `externalId` is generated from the API
+
+### Removed
+
+- **[BREAKING]** `generateExternalId` from `form-service`
```
Comment (mymattcarroll) [inline CHANGELOG.md:16]:
```suggestion
- **[BREAKING]** `formService.generateExternalId()`
```

File: src/services/api/external-id.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresAccessRequest: true,
+          },
+        )
+      }
+      case 400:
+      case 404: {
+        throw new OneBlinkAppsError(
+          'We could not find the form your attempting retrieve drafts for. Please contact your administrator to ensure your application configuration has been completed successfully.',
```
Comment (mymattcarroll) [inline src/services/api/external-id.ts:48]:
```suggestion
          'We could not find the form your attempting submit. Please contact your administrator to ensure your application configuration has been completed successfully.',
```

File: src/services/api/external-id.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          {
+            originalError: error,
+            title: 'Error Syncing Drafts',
```
Comment (mymattcarroll) [inline src/services/api/external-id.ts:51]:
```suggestion
            title: 'Error Generating Receipt',
```

File: src/services/api/external-id.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        throw new OneBlinkAppsError(
+          'You do not have access to this form. Please contact your administrator to gain the correct level of access.',
+          {
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresAccessRequest: true,
+          },
+        )
+      }
+      case 400:
```
Comment (mymattcarroll) [inline src/services/api/external-id.ts:45]: Can we please move this one into its own `case` and use the `error.message` as the message.

File: CHANGELOG.md (hunk 1 above)
Comment (jdawg093) [reply CHANGELOG.md:10]: I do this every time I swear...

File: src/services/api/external-id.ts (hunk 3 above)
Comment (jdawg093) [reply src/services/api/external-id.ts:48]: Whoops

File: src/services/api/external-id.ts (hunk 4 above)
Comment (jdawg093) [reply src/services/api/external-id.ts:51]: Whoops x2

File: package.json [hunk 6]
```diff
@@ -19,15 +19,14 @@
...
-    "nanoid": "^5.0.2",
     "ua-parser-js": "^1.0.37",
     "uuid": "^9.0.1"
   },
   "devDependencies": {
     "@babel/plugin-transform-modules-commonjs": "^7.23.0",
     "@microsoft/eslint-plugin-sdl": "^0.2.0",
     "@oneblink/release-cli": "^3.2.2",
-    "@oneblink/types": "github:oneblink/types",
+    "@oneblink/types": "github:oneblink/types#7517a73f920c21527de7bb01792823d66d5f47e5",
```
Comment (mymattcarroll) [inline package.json:29]: Needs updating
